pyscriptjs/src/pyexec.ts

```diff
@@ -70,17 +70,26 @@ export function displayPyException(err: Error, errElem: HTMLElement) {
     //addClasses(errElem, ['py-error'])
     const pre = document.createElement('pre');
     pre.className = 'py-error';
-
-    if (err.name === 'PythonError') {
-        // err.message contains the python-level traceback (i.e. a string
-        // starting with: "Traceback (most recent call last) ..."
-        logger.error('Python exception:\n' + err.message);
-        pre.innerText = err.message;
-    } else {
-        // this is very likely a normal JS exception. The best we can do is to
-        // display it as is.
-        logger.error('Non-python exception:\n' + err.toString());
-        pre.innerText = err.toString();
+    let errorCount = 0;
+    const errorMsg = err.name === 'PythonError' ? err.message : err.toString();
+    const pyErrorElements = document.getElementsByClassName('py-error');
+    for (const pyErrorElement of pyErrorElements) {
+        if ((pyErrorElement as HTMLElement).innerText.includes(errorMsg)) {
+            errorCount++;
```

Comment:
the only useful case in this code seems to be when `errorCount` is `0` because in the next `if` there's no `else`. Accordingly, wouldn't it be better to remove this flag completely and just `return` from this function doing nothing whenever a previously shown error is already on the page?

On the other side, and this might be about *DX*, wouldn't it be better if the error that already happened *flashes* somehow to underline it's the same error over and over?

Last, but not least: if it's the same error we're after, shouldn't we just check for equality instead of checking if the current text includes the error message?

Thanks in advance for possible clarification around this!

Reply:
As an alternative UI suggestion, `errorCount` could be used to augment the display of the errorBanner itself, so the user has visibility into how many times the error has occurred. Which might be useful if the error occurs, say, many times over due to lots of similar events being triggered at the same time or in close succession.

Reply:
> Last, but not least: if it's the same error we're after, shouldn't we just check for equality instead of checking if the current text includes the error message?
> 
Apologies @WebReflection, it does need to be `equality` instead of `includes`, I will make that change.

Regarding `errorCount`, will it be more better as per @JeffersGlass suggesstion to use the `errorCount` and display the same in the error banner to let user know how many times the error has occurred?

Also regarding this,

> the only useful case in this code seems to be when `errorCount` is `0` because in the next `if` there's no `else`. Accordingly, wouldn't it be better to remove this flag completely and just `return` from this function doing nothing whenever a previously shown error is already on the page?

This change would make more sense right?
```js
// logic to find error count
if (errorCount > 0) {
    return;
}
// logic to render error elements
```

Reply:
@JeffersGlass @aneesh98 I am not sure I am following here because the whole point is that if the error is the same it won't be attached to the page, meaning there could be one and only one specific error showed per time, so that counter can be either `0` or `1` and never more than `1`, which is why I've suggested to remove it.

On the other side, I like the suggestion but with the current logic I'm afraid I don't see it working ... basically whenever we find the same error already happened either we store that counter with the component or we'll show `1` forever ... so how about we use `pyErrorElement.dataset.counter = (parseInt(pyErrorElement.dataset.counter) || 1) + 1` and we use css to show such incremental error counter?

Would this work?

Reply:
Hi @WebReflection, let me know if I get this right :sweat_smile:? You mean to store the errorCount in attribute in `pyErrorElement` compoenent as a state and use css to display the count?

Reply:
Got it! Thanks